middleware/kubernetes/kubernetes.go

```diff
@@ -92,13 +101,27 @@ func (k *Kubernetes) Services(state request.Request, exact bool, opt middleware.
 		return nil, nil, e
 	}
 
+	if state.Name() == DefaultNSName+"."+r.zone && state.Type() == "A" {
```

Comment:
Do we really need to do the additions with "." and r.zone or can we just do strings.Index(state.Name(), DefaultNSName) == 0 ?